### PR TITLE
chore(typescript): extract handlers types

### PR DESCRIPTION
**Pull request by leonardobazico**
I've been rewriting the handlers' types on my current project. To avoid it this PR extract handlers type and expose them

Also made the callbacks have the same types implemented on the [http-proxy](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/51504fd999031b7f025220fab279f1b2155cbaff/types/http-proxy/index.d.ts)

### CODE REVIEW

**Comment by chimurai** [comment]
Thanks for the PR.

Could you add some description and motivation for this change?

**Comment by leonardobazico** [comment]
@chimurai done. Sorry for missed that

**Comment by chimurai** [comment]
Thanks for the improvement!

**Comment by leonardobazico** [comment]
Thank you @chimurai

**Comment by chimurai** [comment]
published in `v2.0.1` (bit late)

Thanks for the contribution!

File: src/types.ts [hunk 1]
```diff
@@ -54,3 +49,41 @@ interface LogProvider {
...
+ */
+export type OnErrorCallback = (
+  err: Error,
+  req: http.IncomingMessage,
+  res: http.ServerResponse,
+  target?: string | Partial<url.Url>
+) => void;
+export type OnProxyResCallback = (
+  proxyRes: http.IncomingMessage,
+  req: http.IncomingMessage,
```
Comment (elado) [inline src/types.ts:67]: Shouldn't this remain `Request` (`express.Request`)?

**Pull request merged**